src/modules/rover_pos_control/RoverPositionControl.cpp

```diff
@@ -501,13 +501,18 @@ RoverPositionControl::run()
 			//orb_copy(ORB_ID(vehicle_attitude), _vehicle_attitude_sub, &_vehicle_att);
 			_act_controls.timestamp = hrt_absolute_time();
 
-			if (_actuator_controls_pub != nullptr) {
-				//PX4_INFO("Publishing actuator from pos control");
-				orb_publish(ORB_ID(actuator_controls_0), _actuator_controls_pub, &_act_controls);
+			/* Only publish if any of the proper modes are enabled */
+			if (_control_mode.flag_control_velocity_enabled ||
+			    manual_mode) {
+				/* publish the actuator controls */
+				if (_actuator_controls_pub != nullptr) {
```

Comment:
How about using "orb_publish_auto" function to handle nullptr issues?

Reply:
Right, or even better would be to use the new API:
https://github.com/PX4/Firmware/blob/6369ae858a79cc410c1963c1c0046bfd8a59f2a7/src/modules/uORB/Publication.hpp#L68

@jbeyerstedt could you do that?

Reply:
@julianoes There are cases scattered around the codebase that passes handle and meta everytime the publishing is needed. Like for example ultrasonic sensor distanve sensor driver. The Publication API really tidies up the code. I guess the left overs were from the previous era without such class / API

Reply:
Now I use the publication API. For the sake of completeness, I also moved the `position_controller_status` publication to the new API. But most of the subscriptions still use the old one (but that's not really part of this PR).